pyccel/ast/utilities.py

```diff
@@ -218,7 +218,8 @@ def builtin_import(expr):
     if source == 'pyccel.decorators':
         funcs = [f[0] for f in inspect.getmembers(pyccel_decorators, inspect.isfunction)]
         for target in expr.target:
-            if str(target) not in funcs:
+            tmp = str(target).split()
+            if tmp[0] not in funcs:
```

Comment:
Can you give an example where split is required?

Reply:
for example:

`from pyccel.decorators import types as func_types`

` types as func_types` is considered as an import

Reply:
Hmm, I think this would be clearer if you used:
```
target = target.target if isinstance(target, AsName) else target
```

Reply:
this will make pyccel look for `func_types` in `pyccel.decorators`

in order to check for the imports properly i need to build an import dictionary which i will use to check if the import is defined or not.

Reply:
> this will make pyccel look for `func_types` in `pyccel.decorators`
Ah, sorry then you should use:
`search_target = target.name if isinstance(target, AsName) else target`
> in order to check for the imports properly i need to build an import dictionary which i will use to check if the import is defined or not.
I am not sure I understand why this is necessary

Reply:
ah sorry its not necessary.